torchtune/data/_messages.py

```diff
@@ -176,12 +185,24 @@ class InputOutputToMessages(Transform):
 
     def __init__(
         self,
-        train_on_input: bool = False,
+        masking_strategy: Optional[str] = "train_on_all",
```

Comment:
if the default value for `train_on_input` was False, this would correspond to `train_on_assistant`. Let's keep that same default behavior.

Reply:
Updated default strategy